rpc/handlers.go

```diff
@@ -746,8 +746,13 @@ func (h *Handler) Events(args *EventsArg) (*EventsChunk, *jsonrpc.Error) {
 
 	emittedEvents := make([]*EmittedEvent, 0, len(filteredEvents))
 	for _, fEvent := range filteredEvents {
+
+		var blockNumber *uint64
+		if fEvent.BlockHash != nil {
+			blockNumber = &fEvent.BlockNumber
+		}
 		emittedEvents = append(emittedEvents, &EmittedEvent{
-			BlockNumber:     fEvent.BlockNumber,
+			BlockNumber:     blockNumber,
 			BlockHash:       fEvent.BlockHash,
```

Comment:
Should this also be nil for pending event?

Reply:
the trick we have to do for BlockNumber is only needed with fEvent.BlockNumber is a value not a pointer. fEvent.BlockHash is a `*felt.Felt` already and will be nil for pending events.